src/Chapter3.hs

```diff
@@ -374,6 +380,37 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Knight = MkKnight
+    { knightHealth :: Int
+    , knightAttack :: Int
+    , knightGold   :: Int
+    }
+
+data Monster = MkMonster
+    { monsterHealth :: Int
+    , monsterAttack :: Int
+    , monsterGold   :: Int
+    }
+
+data FightRound = KnightRound | MonsterRound
+
+fight :: Knight -> Monster -> Int
+fight k m = hit k m KnightRound
+    where
+        hit :: Knight -> Monster -> FightRound -> Int
+        -- knight => monster
+        hit knight monster KnightRound =
```

Comment:
Nice switcher idea!

You could also write the `nextRound ::FightRound -> FightRound` function to help you with the next round. In that case you would be able to remove a bit of the duplicated code 🙂